src/Illuminate/Collection/Collection.php

```diff
@@ -0,0 +1,1766 @@
+<?php
+
+namespace Illuminate\Collection;
+
+use stdClass;
+use Countable;
+use Exception;
+use ArrayAccess;
+use Traversable;
+use ArrayIterator;
+use CachingIterator;
+use JsonSerializable;
+use IteratorAggregate;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Debug\Dumper;
+use Illuminate\Support\Traits\Macroable;
+use Illuminate\Contracts\Support\Jsonable;
+use Illuminate\Contracts\Support\Arrayable;
```

Comment:
So technically `illuminate/collection` will have to require `illuminate/support` and `illuminate/contracts`, how would this make any different from how it is previously?

Reply:
In theory and practice, no. But, uncoupling from `illuminate/support` will clean some code, and be placed under its own namespace.

Reply:
I don't really see how it "cleans up" anything tbh.